File: creational/factory-method/python/concrete_product.py
from product import Logger


class ConsoleLogger(Logger):
    def log(self, message):
        print(f"[Console Logger] {message}")


class FileLogger(Logger):
    def log(self, message):
        with open("log.txt", "a") as file:
            file.write(f"[File Logger] {message}\n")

File: creational/factory/typescript/concrete-product.ts
import * as fs from 'fs';
import { Logger } from './product';

export class ConsoleLogger implements Logger {
	log(message: string) {
		console.log(message);
	}
}

export class FileLogger implements Logger {
	log(message: string) {
		fs.appendFileSync('log.txt', message + '\n');
	}
}
